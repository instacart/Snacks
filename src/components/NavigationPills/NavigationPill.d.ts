import * as React from 'react'
import { WithThemeInjectedProps, ApplyWithTheme } from '../../styles/themer/withTheme'
import { ElementAttributes } from '../..'

export interface NavigationPillProps extends WithThemeInjectedProps {
  /** Any additonal props to add to the element (e.g. data attributes). */
  elementAttributes?: ElementAttributes<'li'>

  /** Any additonal props to add to the inner a element (e.g. data attributes). */
  anchorItemAttributes: ElementAttributes<'a'>

  /** determines wether or not active styles are applied */
  isActive?: boolean

  /** Callback function called after pill click
   * @param event The react `SyntheticEvent`
   * @param props All the props passed to the component
   */
  onClick(e: React.MouseEvent<HTMLElement>, props: NavigationPillProps): void

  /** url used for href property */
  path?: string

  /** text to appear inside pill */
  text?: string
}

declare const NavigationPill: ApplyWithTheme<React.ComponentClass<NavigationPillProps>>

export default NavigationPill
