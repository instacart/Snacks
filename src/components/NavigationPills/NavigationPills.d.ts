import * as React from 'react'
import { WithThemeInjectedProps, ApplyWithTheme } from '../../styles/themer/withTheme'
import { ElementAttributes } from '../..'
import { NavigationPillProps } from './NavigationPill'

interface Pill
  extends Pick<NavigationPillProps, 'text' | 'elementAttributes' | 'anchorItemAttributes'> {}

export interface NavigationPillsProps extends WithThemeInjectedProps {
  /** Any additonal props to add to the element (e.g. data attributes). */
  elementAttributes?: ElementAttributes<'div'>

  /** Any additonal props to add to the inner ul element (e.g. data attributes). */
  listItemAttributes: ElementAttributes<'ul'>

  /** array of pill objects */
  pills: Pill[]

  /**
   * Callback function called after pill click
   * @param e The react `SyntheticEvent`
   * @param props All the props passed to the component
   */
  onClick(e: React.MouseEvent<HTMLElement>, props: Pill): void

  /** optional label placed in front of pills */
  label?: string

  /** string matching the text of one of the pills. Determines which pill is active, if any */
  activePill?: string

  /** optional string for overriding the tag name of the label which appears in front of pills */
  labelElementType?: keyof JSX.IntrinsicElements
}

declare const NavigationPills: ApplyWithTheme<React.ComponentClass<NavigationPillsProps>>

export default NavigationPills
