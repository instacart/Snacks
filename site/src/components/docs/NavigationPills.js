import React, {Fragment} from 'react'
import Playground from '../Playground'

export default () => (

NavigationPills example:

<Playground>
  {`

    initialState = { activePill: 'carrots' };
    pills = [
      { text: 'bananas' },
      { text: 'carrots' },
      { text: 'watermelon' },
      { text: 'snacks' },
      { text: 'kale' },
      { text: 'endives' },
      { text: 'arugula' },
      { text: 'spinach' },
      { text: 'potatoes' }
    ];
    <div>
      <NavigationPills
        pills={pills}
        onPillClick={(e, pill) => {
          e.preventDefault();
          setState({ activePill: pill.text })
          console.log('NavigationPills NavigationPill clicked!', pill)
        }}
        label={'Filter by'}
        activePill={state.activePill}
      />
    </div>
  `}
</Playground>

)
