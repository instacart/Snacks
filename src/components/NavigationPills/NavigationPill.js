import React              from 'react'
import PropTypes          from 'prop-types'
import Radium             from 'radium'
import colors             from '../../styles/colors'
import withTheme          from '../../styles/themer/withTheme'
import { themePropTypes } from '../../styles/themer/utils'

const styles = {
  container: {
    display: 'inline-block'
  },
  main: {
    default: {
      padding: '12px 16px',
      display: 'block',
      fontSize: '14px',
      backgroundColor: colors.GRAY_97,
      borderRadius: '24px',
      margin: '0 4px',
      lineHeight: '1.2',
      transition: 'background-color 150ms ease-in-out',

      ':hover': {
        textDecoration: 'none',
        backgroundColor: colors.GRAY_93
      },

      ':focus': {
        textDecoration: 'none',
        backgroundColor: colors.GRAY_93,
        outline: 'none'
      }
    }
  }
}

const NavigationPill = props => {
  const { isActive, snacksTheme, text } = props
  const { primaryForeground } = snacksTheme.colors

  const activeStyles = {
    backgroundColor: primaryForeground,
    color: colors.WHITE,

    ':hover': {
      backgroundColor: primaryForeground
    },
    ':focus': {
      backgroundColor: primaryForeground,
    }
  }

  return (
    <li style={styles.container} role='tab'>
      <a
        href={props.path || '#'}
        data-bypass={true}
        onClick={e => props.onClick(e, props)}
        style={[
          styles.main.default,
          { color: primaryForeground },
          isActive && activeStyles
        ]}
        key={`pill-anchor-${text}`}
      >
        {text}
      </a>
    </li>
  )
}

NavigationPill.propTypes = {
  /** determines wether or not active styles are applied */
  isActive: PropTypes.bool,
  /** Callback function called after pill click
   * @param {SyntheticEvent} event The react `SyntheticEvent`
   * @param {props} object All the props passed to the component
  */
  onClick: PropTypes.func,
  /** url used for href property */
  path: PropTypes.string,
  /** snacks theme attributes */
  snacksTheme: themePropTypes,
  /** text to appear inside pill */
  text: PropTypes.string
}

NavigationPill.defaultProps = {
  isActive: false
}

export default withTheme(Radium(NavigationPill))
