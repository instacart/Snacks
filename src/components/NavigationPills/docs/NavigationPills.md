NavigationPills example:

```jsx static
const initialState = { activePill: 'carrots' }
const pills = [
  { text: 'bananas' },
  { text: 'carrots' },
  { text: 'watermelon' },
  { text: 'snacks' },
  { text: 'kale' },
  { text: 'endives' },
  { text: 'arugula' },
  { text: 'spinach' },
  { text: 'potatoes' },
]

<div>
  <NavigationPills
    pills={pills}
    onPillClick={(e, pill) => {
      e.preventDefault()
      setState({ activePill: pill.text })
      console.log('NavigationPills NavigationPill clicked!', pill)
    }}
    label="Filter by"
    activePill={state.activePill}
  />
</div>
```
