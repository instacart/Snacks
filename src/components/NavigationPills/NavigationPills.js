import React           from 'react'
import PropTypes       from 'prop-types'
import Radium          from 'radium'
import NavigationPill  from './NavigationPill'
import ScrollTrack     from '../ScrollTrack/ScrollTrack'
import colors          from '../../styles/colors'
import { spacings }    from '../../styles/spacing'

const styles = {
  labelStyles: {
    marginRight: '10px',
  },
  wrapperStyles: {
    display: 'inline-block',
    height: '56px',
    minWidth: '100%',
    padding: `${spacings.XS}px ${spacings.XS}px ${spacings.XS}px ${spacings.MD}px`,
    backgroundColor: colors.WHITE,
    boxSizing: 'border-box'
  },
  pillsContainerStyles: {
    display: 'inline-block',
    margin: '0'
  }
}

const NavigationPills = props => {
  const renderLabel = () => {
    if (!props.label) { return }

    return <label style={styles.labelStyles}>{props.label}</label>
  }

  const renderPill = (pill, idx) => {
    return (
      <NavigationPill
        isActive={props.activePill === pill.text}
        onClick={e => props.onPillClick(e, pill)}
        text={pill.text}
        key={`pill-${idx}`}
      />
    )
  }

  const { pillsContainerStyles, wrapperStyles } = styles

  if (!props.pills || props.pills.length <= 1) { return null }

  return (
    <ScrollTrack>
      <div
        style={wrapperStyles}
        ref='pillsTrack'
      >
        {renderLabel()}
        <ul style={pillsContainerStyles} role='tablist'>
          {props.pills.map(renderPill)}
        </ul>
      </div>
    </ScrollTrack>
  )
}

NavigationPills.propTypes = {
  /** array of pill objects */
  pills: PropTypes.array,
  /** Callback function called after pill click
   * @param {SyntheticEvent} event The react `SyntheticEvent`
   * @param {props} object All the props passed to the component
  */
  onPillClick: PropTypes.func,
  /** optional label placed in front of pills */
  label: PropTypes.string,
  /** string matching the text of one of the pills. Determines which pill is active, if any */
  activePill: PropTypes.string
}

export default Radium(NavigationPills)
