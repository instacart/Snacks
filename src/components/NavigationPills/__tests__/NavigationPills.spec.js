import React from 'react'
import renderer from 'react-test-renderer'
import { StyleRoot } from '@instacart/radium'
import { mount } from 'enzyme'
import { spy } from 'sinon'
import NavigationPills from '../NavigationPills'

const testPills = [
  { path: '/store/doms-pizza-store', text: 'doms' },
  { text: 'doms1' },
  { path: '/store/doms-pizza-store-two', text: 'doms2' },
  { path: '/store/doms-pizza-store-three', text: 'doms3' },
  { path: '/store/doms-pizza-store-four' },
]

it('renders NavigationPills correctly', () => {
  const tree = renderer
    .create(
      <StyleRoot>
        <div>
          <NavigationPills
            pills={testPills}
            onPillClick={(e, pill) => {
              console.log(pill)
            }}
            label={'Filter by'}
            activePill={'dom2'}
          />
        </div>
      </StyleRoot>
    )
    .toJSON()
  expect(tree).toMatchSnapshot()
})

it('renders NavigationPills wihtout label correctly', () => {
  const tree = renderer
    .create(
      <StyleRoot>
        <div>
          <NavigationPills
            pills={testPills}
            onPillClick={(e, pill) => {
              console.log(pill)
            }}
            activePill={'dom2'}
          />
        </div>
      </StyleRoot>
    )
    .toJSON()
  expect(tree).toMatchSnapshot()
})

it('renders NavigationPills without pills correctly', () => {
  const tree = renderer
    .create(
      <StyleRoot>
        <div>
          <NavigationPills
            onPillClick={(e, pill) => {
              console.log(pill)
            }}
            activePill={'dom2'}
            label={'Filter by'}
          />
        </div>
      </StyleRoot>
    )
    .toJSON()
  expect(tree).toMatchSnapshot()
})

it('renders NavigationPills with elementAttributes correctly', () => {
  const tree = renderer
    .create(
      <StyleRoot>
        <div>
          <NavigationPills
            pills={testPills}
            onPillClick={(e, pill) => {
              console.log(pill)
            }}
            label={'Filter by'}
            activePill={'dom2'}
            elementAttributes={{ ariaLabel: 'this is an aria label', role: 'tabs' }}
          />
        </div>
      </StyleRoot>
    )
    .toJSON()
  expect(tree).toMatchSnapshot()
})

it('renders NavigationPills with inner elementAttributes correctly', () => {
  const tree = renderer
    .create(
      <StyleRoot>
        <div>
          <NavigationPills
            pills={testPills}
            onPillClick={(e, pill) => {
              console.log(pill)
            }}
            label={'Filter by'}
            activePill={'dom2'}
            elementAttributes={{ role: 'tabs' }}
            listItemAttributes={{ ariaLabel: 'this is an aria label' }}
          />
        </div>
      </StyleRoot>
    )
    .toJSON()
  expect(tree).toMatchSnapshot()
})

it("renders NavigationPills with each pill's elementAttributes correctly", () => {
  const testPillsElmAttrs = [
    {
      path: '/store/doms-pizza-store',
      text: 'doms',
      elementAttributes: { 'aria-label': 'doms' },
      anchorItemAttributes: { role: 'tab' },
    },
    {
      text: 'doms1',
      elementAttributes: { 'aria-label': 'doms1' },
      anchorItemAttributes: { role: 'tab' },
    },
    {
      path: '/store/doms-pizza-store-two',
      text: 'doms2',
      elementAttributes: { 'aria-label': 'doms2' },
      anchorItemAttributes: { role: 'tab' },
    },
    {
      path: '/store/doms-pizza-store-three',
      text: 'doms3',
      elementAttributes: { 'aria-label': 'doms3' },
      anchorItemAttributes: { role: 'tab' },
    },
    {
      path: '/store/doms-pizza-store-four',
      elementAttributes: { 'aria-label': 'four' },
      anchorItemAttributes: { role: 'tab' },
    },
  ]

  const tree = renderer
    .create(
      <StyleRoot>
        <div>
          <NavigationPills
            pills={testPillsElmAttrs}
            onPillClick={(e, pill) => {
              console.log(pill)
            }}
            label={'Filter by'}
            activePill={'dom2'}
            elementAttributes={{ ariaLabel: 'this is an aria label', role: 'tabs' }}
          />
        </div>
      </StyleRoot>
    )
    .toJSON()
  expect(tree).toMatchSnapshot()
})

it('renders NavigationPills with custom labelElementType', () => {
  const onPillClick = spy()
  const Pills = mount(
    <StyleRoot>
      <div>
        <NavigationPills
          pills={testPills}
          onPillClick={onPillClick}
          label={'Filter by'}
          activePill={'dom2'}
          labelElementType={'h2'}
        />
      </div>
    </StyleRoot>
  )

  const h2 = Pills.find('h2')
  expect(h2.length).toEqual(1)
  expect(h2.first().text()).toEqual('Filter by')
})

it('it handles onPillClick correctly', () => {
  const onPillClick = spy()
  const Pills = mount(
    <StyleRoot>
      <div>
        <NavigationPills
          pills={testPills}
          onPillClick={onPillClick}
          label={'Filter by'}
          activePill={'dom2'}
        />
      </div>
    </StyleRoot>
  )

  const pill = Pills.find('li a').first()
  pill.simulate('click')
  expect(onPillClick.calledOnce).toBeTruthy()
  expect(onPillClick.withArgs(2)).toBeTruthy()
})
