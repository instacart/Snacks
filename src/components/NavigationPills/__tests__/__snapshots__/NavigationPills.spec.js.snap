// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders NavigationPills correctly 1`] = `
<div
  data-radium={true}
>
  <div>
    <div
      data-radium={true}
      style={
        Object {
          "overflow": "hidden",
          "position": "relative",
        }
      }
    >
      <div
        className="rmq-16b35912"
        data-radium={true}
        style={
          Object {
            "left": 0,
            "position": "relative",
            "transition": "left 150ms ease-in-out",
            "whiteSpace": "nowrap",
          }
        }
      >
        <div
          data-radium={true}
          style={Object {}}
        >
          <div
            data-radium={true}
            style={
              Object {
                "backgroundColor": "#FFFFFF",
                "boxSizing": "border-box",
                "display": "inline-block",
                "height": "56px",
                "minWidth": "100%",
                "padding": "8px 8px 8px 24px",
              }
            }
          >
            <label
              data-radium={true}
              style={
                Object {
                  "marginRight": "10px",
                }
              }
            >
              Filter by
            </label>
            <ul
              data-radium={true}
              style={
                Object {
                  "display": "inline-block",
                  "margin": "0",
                }
              }
            >
              <li
                data-radium={true}
                style={
                  Object {
                    "display": "inline-block",
                  }
                }
              >
                <a
                  data-bypass={true}
                  data-radium={true}
                  href="#"
                  onBlur={[Function]}
                  onClick={[Function]}
                  onFocus={[Function]}
                  onMouseEnter={[Function]}
                  onMouseLeave={[Function]}
                  style={
                    Object {
                      "backgroundColor": "#F7F7F7",
                      "borderRadius": "24px",
                      "color": "#43B02A",
                      "display": "block",
                      "fontSize": "14px",
                      "lineHeight": "1.2",
                      "margin": "0 4px",
                      "padding": "12px 16px",
                      "transition": "background-color 150ms ease-in-out",
                    }
                  }
                >
                  doms
                </a>
              </li>
              <li
                data-radium={true}
                style={
                  Object {
                    "display": "inline-block",
                  }
                }
              >
                <a
                  data-bypass={true}
                  data-radium={true}
                  href="#"
                  onBlur={[Function]}
                  onClick={[Function]}
                  onFocus={[Function]}
                  onMouseEnter={[Function]}
                  onMouseLeave={[Function]}
                  style={
                    Object {
                      "backgroundColor": "#F7F7F7",
                      "borderRadius": "24px",
                      "color": "#43B02A",
                      "display": "block",
                      "fontSize": "14px",
                      "lineHeight": "1.2",
                      "margin": "0 4px",
                      "padding": "12px 16px",
                      "transition": "background-color 150ms ease-in-out",
                    }
                  }
                >
                  doms1
                </a>
              </li>
              <li
                data-radium={true}
                style={
                  Object {
                    "display": "inline-block",
                  }
                }
              >
                <a
                  data-bypass={true}
                  data-radium={true}
                  href="#"
                  onBlur={[Function]}
                  onClick={[Function]}
                  onFocus={[Function]}
                  onMouseEnter={[Function]}
                  onMouseLeave={[Function]}
                  style={
                    Object {
                      "backgroundColor": "#F7F7F7",
                      "borderRadius": "24px",
                      "color": "#43B02A",
                      "display": "block",
                      "fontSize": "14px",
                      "lineHeight": "1.2",
                      "margin": "0 4px",
                      "padding": "12px 16px",
                      "transition": "background-color 150ms ease-in-out",
                    }
                  }
                >
                  doms2
                </a>
              </li>
              <li
                data-radium={true}
                style={
                  Object {
                    "display": "inline-block",
                  }
                }
              >
                <a
                  data-bypass={true}
                  data-radium={true}
                  href="#"
                  onBlur={[Function]}
                  onClick={[Function]}
                  onFocus={[Function]}
                  onMouseEnter={[Function]}
                  onMouseLeave={[Function]}
                  style={
                    Object {
                      "backgroundColor": "#F7F7F7",
                      "borderRadius": "24px",
                      "color": "#43B02A",
                      "display": "block",
                      "fontSize": "14px",
                      "lineHeight": "1.2",
                      "margin": "0 4px",
                      "padding": "12px 16px",
                      "transition": "background-color 150ms ease-in-out",
                    }
                  }
                >
                  doms3
                </a>
              </li>
              <li
                data-radium={true}
                style={
                  Object {
                    "display": "inline-block",
                  }
                }
              >
                <a
                  data-bypass={true}
                  data-radium={true}
                  href="#"
                  onBlur={[Function]}
                  onClick={[Function]}
                  onFocus={[Function]}
                  onMouseEnter={[Function]}
                  onMouseLeave={[Function]}
                  style={
                    Object {
                      "backgroundColor": "#F7F7F7",
                      "borderRadius": "24px",
                      "color": "#43B02A",
                      "display": "block",
                      "fontSize": "14px",
                      "lineHeight": "1.2",
                      "margin": "0 4px",
                      "padding": "12px 16px",
                      "transition": "background-color 150ms ease-in-out",
                    }
                  }
                />
              </li>
            </ul>
          </div>
        </div>
      </div>
    </div>
  </div>
  <style
    dangerouslySetInnerHTML={
      Object {
        "__html": "@media (max-width: 767px){ .rmq-16b35912{left: inherit !important;
      overflow-x: scroll !important;
      display: block !important;
      -ms-overflow-style: none !important;
      overflow: -moz-scrollbars-none !important;
      webkit-overflow-scrolling: touch !important;}}",
      }
    }
  />
</div>
`;

exports[`renders NavigationPills wihtout label correctly 1`] = `
<div
  data-radium={true}
>
  <div>
    <div
      data-radium={true}
      style={
        Object {
          "overflow": "hidden",
          "position": "relative",
        }
      }
    >
      <div
        className="rmq-16b35912"
        data-radium={true}
        style={
          Object {
            "left": 0,
            "position": "relative",
            "transition": "left 150ms ease-in-out",
            "whiteSpace": "nowrap",
          }
        }
      >
        <div
          data-radium={true}
          style={Object {}}
        >
          <div
            data-radium={true}
            style={
              Object {
                "backgroundColor": "#FFFFFF",
                "boxSizing": "border-box",
                "display": "inline-block",
                "height": "56px",
                "minWidth": "100%",
                "padding": "8px 8px 8px 24px",
              }
            }
          >
            <ul
              data-radium={true}
              style={
                Object {
                  "display": "inline-block",
                  "margin": "0",
                }
              }
            >
              <li
                data-radium={true}
                style={
                  Object {
                    "display": "inline-block",
                  }
                }
              >
                <a
                  data-bypass={true}
                  data-radium={true}
                  href="#"
                  onBlur={[Function]}
                  onClick={[Function]}
                  onFocus={[Function]}
                  onMouseEnter={[Function]}
                  onMouseLeave={[Function]}
                  style={
                    Object {
                      "backgroundColor": "#F7F7F7",
                      "borderRadius": "24px",
                      "color": "#43B02A",
                      "display": "block",
                      "fontSize": "14px",
                      "lineHeight": "1.2",
                      "margin": "0 4px",
                      "padding": "12px 16px",
                      "transition": "background-color 150ms ease-in-out",
                    }
                  }
                >
                  doms
                </a>
              </li>
              <li
                data-radium={true}
                style={
                  Object {
                    "display": "inline-block",
                  }
                }
              >
                <a
                  data-bypass={true}
                  data-radium={true}
                  href="#"
                  onBlur={[Function]}
                  onClick={[Function]}
                  onFocus={[Function]}
                  onMouseEnter={[Function]}
                  onMouseLeave={[Function]}
                  style={
                    Object {
                      "backgroundColor": "#F7F7F7",
                      "borderRadius": "24px",
                      "color": "#43B02A",
                      "display": "block",
                      "fontSize": "14px",
                      "lineHeight": "1.2",
                      "margin": "0 4px",
                      "padding": "12px 16px",
                      "transition": "background-color 150ms ease-in-out",
                    }
                  }
                >
                  doms1
                </a>
              </li>
              <li
                data-radium={true}
                style={
                  Object {
                    "display": "inline-block",
                  }
                }
              >
                <a
                  data-bypass={true}
                  data-radium={true}
                  href="#"
                  onBlur={[Function]}
                  onClick={[Function]}
                  onFocus={[Function]}
                  onMouseEnter={[Function]}
                  onMouseLeave={[Function]}
                  style={
                    Object {
                      "backgroundColor": "#F7F7F7",
                      "borderRadius": "24px",
                      "color": "#43B02A",
                      "display": "block",
                      "fontSize": "14px",
                      "lineHeight": "1.2",
                      "margin": "0 4px",
                      "padding": "12px 16px",
                      "transition": "background-color 150ms ease-in-out",
                    }
                  }
                >
                  doms2
                </a>
              </li>
              <li
                data-radium={true}
                style={
                  Object {
                    "display": "inline-block",
                  }
                }
              >
                <a
                  data-bypass={true}
                  data-radium={true}
                  href="#"
                  onBlur={[Function]}
                  onClick={[Function]}
                  onFocus={[Function]}
                  onMouseEnter={[Function]}
                  onMouseLeave={[Function]}
                  style={
                    Object {
                      "backgroundColor": "#F7F7F7",
                      "borderRadius": "24px",
                      "color": "#43B02A",
                      "display": "block",
                      "fontSize": "14px",
                      "lineHeight": "1.2",
                      "margin": "0 4px",
                      "padding": "12px 16px",
                      "transition": "background-color 150ms ease-in-out",
                    }
                  }
                >
                  doms3
                </a>
              </li>
              <li
                data-radium={true}
                style={
                  Object {
                    "display": "inline-block",
                  }
                }
              >
                <a
                  data-bypass={true}
                  data-radium={true}
                  href="#"
                  onBlur={[Function]}
                  onClick={[Function]}
                  onFocus={[Function]}
                  onMouseEnter={[Function]}
                  onMouseLeave={[Function]}
                  style={
                    Object {
                      "backgroundColor": "#F7F7F7",
                      "borderRadius": "24px",
                      "color": "#43B02A",
                      "display": "block",
                      "fontSize": "14px",
                      "lineHeight": "1.2",
                      "margin": "0 4px",
                      "padding": "12px 16px",
                      "transition": "background-color 150ms ease-in-out",
                    }
                  }
                />
              </li>
            </ul>
          </div>
        </div>
      </div>
    </div>
  </div>
  <style
    dangerouslySetInnerHTML={
      Object {
        "__html": "@media (max-width: 767px){ .rmq-16b35912{left: inherit !important;
      overflow-x: scroll !important;
      display: block !important;
      -ms-overflow-style: none !important;
      overflow: -moz-scrollbars-none !important;
      webkit-overflow-scrolling: touch !important;}}",
      }
    }
  />
</div>
`;

exports[`renders NavigationPills without pills correctly 1`] = `
<div
  data-radium={true}
>
  <div />
  <style
    dangerouslySetInnerHTML={
      Object {
        "__html": "",
      }
    }
  />
</div>
`;
